packages/mobile/src/store/search/searchSagas.ts

```diff
@@ -0,0 +1,40 @@
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import { uniq } from 'lodash'
+import { call, put, select } from 'typed-redux-saga'
+
+import { SEARCH_HISTORY_KEY } from 'app/constants/storage-keys'
+
+import { setHistory } from './searchSlice'
+import { getSearchHistory } from './selectors'
+
+function* syncLegacySearchHistory() {
+  const legacySearchHistoryJson = yield* call(
+    [AsyncStorage, AsyncStorage.getItem],
+    SEARCH_HISTORY_KEY
+  )
+
+  if (!legacySearchHistoryJson) {
+    return
+  }
+
+  const legacySearchHistory: string[] = JSON.parse(legacySearchHistoryJson)
+
+  if (!Array.isArray(legacySearchHistory)) {
+    yield* call([AsyncStorage, AsyncStorage.removeItem], SEARCH_HISTORY_KEY)
+    return
+  }
+
+  const persistedSearchHistory = yield* select(getSearchHistory)
+
+  const mergedSearchHistory = uniq([
+    ...persistedSearchHistory,
```

Comment:
Doesn't really matter, but was it possible to have multiple identical search history items previously? Because this would remove that and make them all unique right

Reply:
yeah good call, i looked at current implementation, and any new item that's the same as a previous one will filter out the old one, and put latest first, so i think this accomplishes that, since persisted queries will be latest, and we filter out the legacy ones if duplicate!